pandas/core/indexing.py

```diff
@@ -1419,7 +1437,17 @@ def _convert_to_indexer(self, key, axis: AxisInt):
         ):
             raise IndexingError("Too many indexers")
 
-        if is_scalar(key) or (isinstance(labels, MultiIndex) and is_hashable(key)):
+        # Slices are not valid keys passed in by the user,
+        # even though they are hashable in Python 3.12
+        contains_slice = False
+        if isinstance(key, tuple):
+            for v in key:
+                if isinstance(v, slice):
+                    contains_slice = True
```

Comment:
`contains_slice = any(isinstance(v, slice) for v in key)`?

Reply:
Thx.